### ISSUE TITLE
Rule proposal: `prefer-destructuring`

### ISSUE THREAD

**Issue by axetroy** [sindresorhus/eslint-plugin-unicorn#2433]
### Description

For clean code, combine multiple declarations of the same method into one

### Fail

```js
const leftCode = sourceCode.getText(left);
const rightCode = sourceCode.getText(right);
const alternateCode = sourceCode.getText(alternate);
const consequentCode = sourceCode.getText(consequent);
```

### Pass

```js
const [leftCode, rightCode, alternateCode, consequentCode] = [left, right, alternate, consequent].map(n => sourceCode.getText(n));
```

### Proposed rule name

prefer-destructuring

**Comment by fregante** [comment]
That looks like a definitely loss of readability and increase of possible bugs. I add a comma and they're _all_ broken.

```js
const [, leftCode, rightCode, alternateCode, consequentCode] = [left, right, alternate, consequent].map(n => sourceCode.getText(n));
```

**Comment by axetroy** [comment]
Make sense.
